### PR TITLE
Add TLS version options to s_time

### PR DESCRIPTION
**Pull request by raja-ashok**
Added TLS version options like (`-tls1`, `-tls1_1`, `-tls1_2`, `-tls1_3`) to `s_time`.

##### Checklist

- [x] documentation is added or updated

### CODE REVIEW

File: apps/s_time.c [hunk 1]
```diff
@@ -76,6 +76,10 @@ const OPTIONS s_time_options[] = {
 #ifndef OPENSSL_NO_SSL3
     {"ssl3", OPT_SSL3, '-', "Just use SSLv3"},
 #endif
+    {"tls1", OPT_TLS1, '-', "Just use TLSv1.0"},
```
Comment (slontis) [inline apps/s_time.c]: Just like ssl3 there are OPENSSL_NO_ variants for TLS1 etc..
See apps/ciphers.c for example.

File: apps/s_time.c [hunk 2]
```diff
@@ -179,6 +183,18 @@ int s_time_main(int argc, char **argv)
         case OPT_SSL3:
             max_version = SSL3_VERSION;
             break;
+        case OPT_TLS1:
+            max_version = TLS1_VERSION;
```
Comment (slontis) [inline apps/s_time.c]: Wouldn't you also need to set min_version?

File: doc/man1/openssl-s_time.pod [hunk 3]
```diff
@@ -22,6 +22,10 @@ B<openssl> B<s_time>
 [B<-nameopt option>]
 [B<-time seconds>]
 [B<-ssl3>]
+[B<-tls1>]
+[B<-tls1_1>]
+[B<-tls1_2>]
+[B<-tls1_3>]
```
Comment (InfoHunter) [inline doc/man1/openssl-s_time.pod]: It seems detailed description of these newly added options also need to be added.

File: apps/s_time.c (hunk 2 above)
Comment (InfoHunter) [reply apps/s_time.c]: Actually I am wondering why we need to introduce the `min_version`, it seems we could just call `SSL_CTX_set_min_proto_version` with the `max_version` variable in this scenario. (or rename it to something like `version`)

File: doc/man1/openssl-s_time.pod [hunk 4]
```diff
@@ -109,19 +113,13 @@ Performs the timing test using the same session ID; this can be used as a test
 that session caching is working. If neither B<-new> nor B<-reuse> are
 specified, they are both on by default and executed in sequence.
 
-=item B<-ssl3>
+=item B<-ssl3>, B<-tls1>, B<-tls1_1>, B<-tls1_2>, B<-tls1_3>
 
-This option disables the use of SSL version 3. By default
```
Comment (mattcaswell) [inline doc/man1/openssl-s_time.pod]: It seems this original text is just wrong!

Comment (raja-ashok) [reply doc/man1/openssl-s_time.pod]: Hmmm. Yes.

**Comment by mattcaswell** [comment]
Pushed. Thanks.

**Pull request closed without merging**